### PR TITLE
Mod3dev

### PR DESCRIPTION
**Pull request by Gradoma**
Migrated to Boot

### CODE REVIEW

File: module #3. REST API Advanced/web/src/main/webapp/index.jsp [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<html>
```
Comment (ilywka) [inline module #3. REST API Advanced/web/src/main/webapp/index.jsp]: remove webapp folder

File: module #3. REST API Advanced/web/pom.xml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.springframework.boot</groupId>
+        <artifactId>spring-boot-maven-plugin</artifactId>
+        <version>2.4.0</version>
+        <configuration>
+          <mainClass>com.epam.esm.Application</mainClass>
+          <layout>WAR</layout>
```
Comment (ilywka) [inline module #3. REST API Advanced/web/pom.xml]: war ?

File: module #3. REST API Advanced/core/src/main/java/com/epam/esm/dao/criteria/CriteriaSet.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
+package com.epam.esm.dao.criteria;
+
+import com.epam.esm.dao.criteria.impl.*;
+
+import java.util.Arrays;
+
+public enum CriteriaSet {
```
Comment (ilywka) [inline module #3. REST API Advanced/core/src/main/java/com/epam/esm/dao/criteria/CriteriaSet.java]: Looks weird. Total number of enum entries is cartesian product of criterias... How can be improved?

**Comment by Gradoma** [comment]
- modify selectors
- spring-boot-maven-plugin - have questions?

File: module #3. REST API Advanced/core/src/main/java/com/epam/esm/service/impl/OrderServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+                            GiftCertificateService giftCertificateService){
+        this.giftCertificateService = giftCertificateService;
+        this.orderDao = orderDao;
+        this.orderDtoMapper = orderDtoMapper;
+    }
+
+    @Override
+    public long add(OrderDto orderDto) {
+        Order order = orderDtoMapper.toEntity(orderDto);
+        GiftCertificateDto certificateDto = giftCertificateService.getById(order.getCertificateId());
```
Comment (Gradoma) [inline module #3. REST API Advanced/core/src/main/java/com/epam/esm/service/impl/OrderServiceImpl.java:34]: it's ok, or better to create separate method for certificate "getCertificatePriceById(long id)" than query full entity just for one field value?

File: module #3. REST API Advanced/web/src/main/java/com/epam/esm/controller/GiftCertificateController.java [hunk 5]
```diff
@@ -62,4 +66,14 @@ public GiftCertificateController(GiftCertificateService giftCertificateService){
...
         return ResponseEntity.noContent().build();
     }
+
+    private void addLinks(GiftCertificateDto certificateDto){
+        certificateDto.getTags().forEach(tagDto -> {
+            long tagId = Long.parseLong(tagDto.getId());
+            Link tagLink = WebMvcLinkBuilder.linkTo(WebMvcLinkBuilder.methodOn(TagController.class)
+                    .getById(tagId))
+                    .withRel("tag");
+            tagDto.add(tagLink);
```
Comment (Gradoma) [inline module #3. REST API Advanced/web/src/main/java/com/epam/esm/controller/GiftCertificateController.java:104]: CertificateDto has all fields of every TagDto. Tag link direct to TagDto with all this fields. It's ok, or show only TagDto id in Certificate, and direct to full info by link?

**Pull request merged**